packages/@azure/servicebus/data-plane/test/testUtils.ts

```diff
@@ -40,13 +42,50 @@ export enum ClientType {
   TopicFilterTestSubscription
 }
 
-export function getSenderClient(
+export async function getSenderClient(
   namespace: Namespace,
   clientType: ClientType
-): QueueClient | TopicClient {
+): Promise<QueueClient | TopicClient> {
   switch (clientType) {
     case ClientType.PartitionedQueue:
-      return namespace.createQueueClient(process.env.QUEUE_NAME || "partitioned-queue");
+      const subscriptionId = process.env["AZURE_SUBSCRIPTION_ID"] || "";
+      const queueName = process.env.QUEUE_NAME || "partitioned-queue";
+      await msRestNodeAuth
+        .loginWithServicePrincipalSecret(
+          process.env.ARM_SERVICEBUS_CLIENT_ID || "",
+          process.env.ARM_SERVICEBUS_SECRET || "",
+          process.env.ARM_SERVICEBUS_TENANT_ID || ""
+        )
+        .then(async (creds) => {
+          const client = await new ServiceBusManagementClient(creds, subscriptionId);
+          await client.queues.deleteMethod(
+            process.env.RESOURCE_GROUP || "",
+            process.env.SERVICEBUS_NAMESPACE || "",
+            queueName,
+            function(error: any): void {
+              if (error) {
+                console.log(error.message);
```

Comment:
Will an error be thrown if the queue we want to delete doesn't exist?

Reply:
Nope, `arm-servicebus` doesn't give any error when we try to delete a non-existing queue. 
 > [Note : _`azure-sb` library allows a similar callback function which provides the error message saying the queue doesn't exist._]

However, `arm-servicebus` gives an error when provided an invalid `RESOURCE_GROUP or SERVICEBUS_NAMESPACE`.